dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -73,7 +69,6 @@
   private final ImmutableList<ClassPathEntry> inputClassPath;
   private final FixedSizeClassPathRepository classRepository;
   private final ClassLoader extensionClassLoader;
-  private final ImmutableSetMultimap<ClassPathEntry, String> classPathEntryToClassFileNames;
```

Comment:
Memo: where does LinkageChecker use this ?

It's used by `classDumper.classNamesInJar()`. This method is used by Linkage Checker:

```
    classToFieldSymbols.forEach(
        (classFile, fieldSymbol) -> {
          if (!classDumper
              .classNamesInJar(classFile.getClassPathEntry())
              .contains(fieldSymbol.getClassBinaryName())) {
            findSymbolProblem(classFile, fieldSymbol)
                .ifPresent(problem -> problemToClass.put(problem, classFile.topLevelClassFile()));
          }
        });
```

This if-statement checks whether the source class and target class of a symbol reference are in the same JAR file or not. If they are in the same JAR file, Linkage Checker skips the references because such javac would not compile to generate such JAR files.

Reply:
Therefore the result of class file name reading should be cached (this PR does that)